From 54bdb25afc82917d05453ebae515d807e5706a9c Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@chromium.org>
Date: Wed, 19 May 2021 11:19:30 +0000
Subject: [PATCH] json_schema_compiler: use python3 for feature_compiler.py

Fixed: 1205587
Change-Id: I31db44af23f9b56725e6a5aecd2c80d6a33cad2f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2905318
Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
Commit-Queue: Nico Weber <thakis@chromium.org>
Reviewed-by: Nico Weber <thakis@chromium.org>
Cr-Commit-Position: refs/heads/master@{#884374}
---
 src/tools/json_schema_compiler/feature_compiler.py | 4 ++--
 src/tools/json_schema_compiler/json_features.gni   | 5 +----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/tools/json_schema_compiler/feature_compiler.py b/src/tools/json_schema_compiler/feature_compiler.py
index 645c5fb934..fab6c6a006 100644
--- a/src/tools/json_schema_compiler/feature_compiler.py
+++ b/src/tools/json_schema_compiler/feature_compiler.py
@@ -377,8 +377,8 @@ def DoesNotHaveAllowlistForHostedApps(value):
   # what the allowlist looks like) to a python list of strings.
   def cpp_list_to_list(cpp_list):
     assert type(cpp_list) is str
-    assert cpp_list[0] is '{'
-    assert cpp_list[-1] is '}'
+    assert cpp_list[0] == '{'
+    assert cpp_list[-1] == '}'
     new_list = json.loads('[%s]' % cpp_list[1:-1])
     assert type(new_list) is list
     return new_list
diff --git a/src/tools/json_schema_compiler/json_features.gni b/src/tools/json_schema_compiler/json_features.gni
index d6d6ac641e..743162e2d5 100644
--- a/src/tools/json_schema_compiler/json_features.gni
+++ b/src/tools/json_schema_compiler/json_features.gni
@@ -2,8 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/python.gni")
-
 # Generates the FeatureProviders files for extension features files.
 # The following variables are required:
 #   sources: The features.json files to use.
@@ -29,8 +27,7 @@ template("json_features") {
     "$target_gen_dir/$base_filename.h",
   ]
 
-  # TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-  python2_action(action_name) {
+  action(action_name) {
     visibility = [ ":$source_set_name" ]
     sources = invoker.sources
     script = "$compiler_root/feature_compiler.py"
